tests/data/dags/test_dag_with_bigquery_table_sensors

```diff
@@ -0,0 +1,93 @@
+# Generated via https://github.com/mozilla/bigquery-etl/blob/main/bigquery_etl/query_scheduling/generate_airflow_dags.py
+
+from airflow import DAG
+from airflow.providers.google.cloud.sensors.bigquery import (
+    BigQueryTableExistenceSensor,
+    BigQueryTablePartitionExistenceSensor,
+)
+from airflow.sensors.external_task import ExternalTaskMarker
+from airflow.sensors.external_task import ExternalTaskSensor
+from airflow.utils.task_group import TaskGroup
+import datetime
+from operators.gcp_container_operator import GKEPodOperator
+from utils.constants import ALLOWED_STATES, FAILED_STATES
+from utils.gcp import bigquery_etl_query, bigquery_dq_check
+
+docs = """
+### bqetl_test_dag
+
+Built from bigquery-etl repo, [`dags/bqetl_test_dag.py`](https://github.com/mozilla/bigquery-etl/blob/generated-sql/dags/bqetl_test_dag.py)
+
+#### Owner
+
+test@example.org
+
+#### Tags
+
+* repo/bigquery-etl
+"""
+
+
+default_args = {
+    "owner": "test@example.org",
+    "start_date": datetime.datetime(2020, 1, 1, 0, 0),
+    "end_date": None,
+    "email": ["test@example.org"],
+    "depends_on_past": False,
+    "retry_delay": datetime.timedelta(seconds=3600),
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+}
+
+tags = ["repo/bigquery-etl"]
+
+with DAG(
+    "bqetl_test_dag",
+    default_args=default_args,
+    schedule_interval="@daily",
+    doc_md=docs,
+    tags=tags,
+) as dag:
+    test__non_incremental_query__v1 = bigquery_etl_query(
+        task_id="test__non_incremental_query__v1",
+        destination_table="non_incremental_query_v1",
+        dataset_id="test",
+        project_id="moz-fx-data-test-project",
+        owner="test@example.com",
+        email=["test@example.com"],
+        date_partition_parameter="submission_date",
+        depends_on_past=True,
+        arguments=["--append_table"],
+    )
+
+    wait_for_foo_bar_baz = BigQueryTableExistenceSensor(
+        task_id="wait_for_foo_bar_baz",
+        project_id="foo",
+        dataset_id="bar",
+        table_id="baz_{{ ds_nodash }}",
```

Comment:
No strong opinions here but if this sensor can be used for sharded tables, can it be used for partitioned tables? So we only need `BigQueryTableExistenceSensor` - something like `baz${{ ds_nodash }}`

Reply:
A clever idea, but I just tried it, and the way the underlying sensor code is written that doesn't work as intended.